From ee92ef53933a9cea5b7a0c0dbb3da480dd56eeef Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 8 May 2019 19:34:48 +0200
Subject: [PATCH 1/1] lib: charset: correct utf8_utf16_strnlen() description

Correct the description of utf8_utf16_strnlen() and utf8_utf16_strlen() to
reflect that they return u16 count and not byte count.

For these functions and utf16_utf8_strnlen() describe the handling of
invalid code sequences.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/charset.h | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/include/charset.h b/include/charset.h
index 65087f76d1..4f7ae8fafd 100644
--- a/include/charset.h
+++ b/include/charset.h
@@ -46,9 +46,9 @@ int utf8_put(s32 code, char **dst);
  *
  * @src:		utf-8 string
  * @count:		maximum number of code points to convert
- * Return:		length in bytes after conversion to utf-16 without the
+ * Return:		length in u16 after conversion to utf-16 without the
  *			trailing \0. If an invalid UTF-8 sequence is hit one
- *			word will be reserved for a replacement character.
+ *			u16 will be reserved for a replacement character.
  */
 size_t utf8_utf16_strnlen(const char *src, size_t count);
 
@@ -56,8 +56,9 @@ size_t utf8_utf16_strnlen(const char *src, size_t count);
  * utf8_utf16_strlen() - length of a utf-8 string after conversion to utf-16
  *
  * @src:		utf-8 string
- * Return:		length in bytes after conversion to utf-16 without the
- *			trailing \0. -1 if the utf-8 string is not valid.
+ * Return:		length in u16 after conversion to utf-16 without the
+ *			trailing \0. If an invalid UTF-8 sequence is hit one
+ *			u16 will be reserved for a replacement character.
  */
 #define utf8_utf16_strlen(a) utf8_utf16_strnlen((a), SIZE_MAX)
 
@@ -127,7 +128,8 @@ size_t utf16_utf8_strnlen(const u16 *src, size_t count);
  *
  * @src:		utf-16 string
  * Return:		length in bytes after conversion to utf-8 without the
- *			trailing \0. -1 if the utf-16 string is not valid.
+ *			trailing \0. If an invalid UTF-16 sequence is hit one
+ *			byte will be reserved for a replacement character.
  */
 #define utf16_utf8_strlen(a) utf16_utf8_strnlen((a), SIZE_MAX)
 
-- 
2.20.1

